src/anomalib/utils/metrics/perimg/pimo.py

```diff
@@ -0,0 +1,211 @@
+"""Per-Image Overlap (PIMO, pronounced pee-mo) curve.
+
+Two variants of AUCs are implemented:
+    - AUPImO: Area Under the Per-Image Overlap (PIMO) curves.
+              I.e. a metric of per-image average TPR.
+
+for shared fpr = mean( perimg fpr ) == set fpr
+    find the th = fpr^-1( MAX_FPR ) with a binary search on the pixels of the norm images
+    i.e. it's not necessary to compute the perimg fpr curves (tf. binclf curves) in advance
+for other shared fpr alternatives, it's necessary to compute the perimg fpr curves first anyway
+
+further: also choose the th upper bound to be the max score at normal pixels
+"""
+
```

Comment:
License header. You could do this later on as well, since this is going into the feature branch.